crates/trigger-http/src/tls.rs

```diff
@@ -19,28 +19,118 @@ impl TlsConfig {
     // Creates a TLS acceptor from server config.
     pub(super) fn server_config(&self) -> anyhow::Result<TlsAcceptor> {
         let certs = load_certs(&self.cert_path)?;
-        let mut keys = load_keys(&self.key_path)?;
+        let private_key = load_keys(&self.key_path)?;
 
         let cfg = rustls::ServerConfig::builder()
-            .with_safe_defaults()
             .with_no_client_auth()
-            .with_single_cert(certs, keys.remove(0))
+            .with_single_cert(certs, private_key)
             .map_err(|e| anyhow::anyhow!("{}", e))?;
 
         Ok(Arc::new(cfg).into())
     }
 }
 
-// Loads public certificate from file.
-fn load_certs(path: impl AsRef<Path>) -> io::Result<Vec<rustls::Certificate>> {
-    certs(&mut io::BufReader::new(fs::File::open(path)?))
-        .map_err(|_| io::Error::new(io::ErrorKind::InvalidInput, "invalid cert"))
-        .map(|mut certs| certs.drain(..).map(rustls::Certificate).collect())
+// load_certs parse and return the certs from the provided file
+pub fn load_certs(
+    path: impl AsRef<Path>,
+) -> io::Result<Vec<rustls_pki_types::CertificateDer<'static>>> {
+    rustls_pemfile::certs(&mut io::BufReader::new(fs::File::open(path).map_err(
+        |err| {
+            io::Error::new(
+                io::ErrorKind::InvalidInput,
+                format!("failed to read cert file {:?}", err),
+            )
+        },
+    )?))
+    .into_iter()
+    .collect::<io::Result<Vec<rustls_pki_types::CertificateDer<'static>>>>()
 }
 
-// Loads private key from file.
-fn load_keys(path: impl AsRef<Path>) -> io::Result<Vec<rustls::PrivateKey>> {
-    pkcs8_private_keys(&mut io::BufReader::new(fs::File::open(path)?))
-        .map_err(|_| io::Error::new(io::ErrorKind::InvalidInput, "invalid key"))
-        .map(|mut keys| keys.drain(..).map(rustls::PrivateKey).collect())
+// load_keys parse and return the first private key from the provided file
+pub fn load_keys(path: impl AsRef<Path>) -> io::Result<rustls_pki_types::PrivateKeyDer<'static>> {
```

Comment:
nit: should this be `load_key` now that it's only loading one key?

Reply:
done.